### PR TITLE
Update preprocessing service for log ingestion

### PR DESCRIPTION
**Pull request by AmartC**
This PR introduces the changes made to the preprocessing service with the introduction of the log ingest plugin from Opensearch now.

### CODE REVIEW

File: preprocessing-service/preprocess.py [hunk 1]
```diff
@@ -63,15 +63,36 @@ async def subscribe_handler(msg):
...
+            await run(payload_data_df, masker)
+
+        start_time = time.time()
+        if start_time - last_time >= 1 or len(pending_list) > 128:
+            payload_data_df = pd.DataFrame(pending_list)
+            last_time = start_time
+            pending_list = []
+            await run(payload_data_df, masker)
+
+async def mask_logs_cpy(queue):
```
Comment (tybalex) [inline preprocessing-service/preprocess.py:85]: what's this?

File: preprocessing-service/preprocess.py [hunk 2]
```diff
@@ -46,15 +47,14 @@ async def doc_generator(df):
...
-        doc_dict["doc"]["anomaly_level"] = "Normal"
         yield doc_dict
 
 
 async def consume_logs(mask_logs_queue):
     async def subscribe_handler(msg):
+
         payload_data = msg.data.decode()
-        await mask_logs_queue.put(pd.read_json(StringIO(payload_data), dtype={"_id": object, "cluster_id": str}))
+        await mask_logs_queue.put(json.loads(payload_data))
```
Comment (sanjay920) [inline preprocessing-service/preprocess.py:57]: this line can't be changed because this was introduced to support the opni support bundle

Comment (sanjay920) [reply preprocessing-service/preprocess.py:57]: it has to be `await mask_logs_queue.put(pd.read_json(StringIO(payload_data), dtype={"_id": object, "cluster_id": str}))`

File: preprocessing-service/preprocess.py [hunk 3]
```diff
@@ -63,108 +63,112 @@ async def subscribe_handler(msg):
...
-            await nw.publish("preprocessed_logs", app_logs_df.to_json().encode())
-        '''
-
-        if len(control_plane_logs_df) > 0:
+        payload_data_df.drop(["_type"], axis=1, errors="ignore", inplace=True)
+        payload_data_df.drop(["_version"], axis=1, errors="ignore", inplace=True)
+        opensearch_pending_list.append(payload_data_df)
+        opensearch_log_count += len(payload_data_df)
+        start_time = time.time()
+        if (start_time - last_time >= 1 and opensearch_log_count > 0) or opensearch_log_count >= 128:
```
Comment (sanjay920) [inline preprocessing-service/preprocess.py:117]: when is `opensearch_log_count` reset?

**Review by sanjay920 — CHANGES_REQUESTED** [review]

File: preprocessing-service/preprocess.py (hunk 3 above)
Comment (AmartC) [reply preprocessing-service/preprocess.py:117]: That is part of an older version. Will remove this function altogether.

File: preprocessing-service/preprocess.py (hunk 2 above)
Comment (tybalex) [reply preprocessing-service/preprocess.py:57]: @sanjay920  why can't we remove it though? if the support bundle can adapt to our latest pipeline, this shouldn't be a problem

**Review by tybalex — APPROVED** [review]

**Review by sanjay920 — APPROVED** [review]

**Pull request merged**